satisfactory-accounting-app/src/node_display/group/group_name.rs

```diff
@@ -124,13 +130,21 @@ impl GroupName {
         let oninput = link.callback(|input| Msg::UpdatePending {
             pending: get_value_from_input_event(input),
         });
+        let onkeydown = link.batch_callback(|e: KeyboardEvent| {
+            if e.key() == "Escape" {
```

Comment:
Sorry, I don't mean to make you do extra changes, but I didn't think to look at this earlier.

Usually I check both "Esc" and "Escape" because some browsers are [documented as returning][1] "Esc" instead of "Escape" for this case. I don't think it's ones that support webassembly in the case of escape, but in the case of arrow keys, it does include a version of Edge that supports webassembly, so I prefer to just cover both cases to be thorough.

[1]: https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values#ui_keys

You can cover both cases and shorten this condition by using a match block, like I do in the case of other parts of the code that handle 'cancel on escape'. [For example here][2].

[2]: https://github.com/satisfactory-accounting/satisfactory-accounting/blob/7a02e97b1cd043d985addfcaf25e235817ebade0/satisfactory-accounting-app/src/node_display/building/clock.rs#L88

```rust
            let onkeyup = link.batch_callback(|e: KeyboardEvent| match &*e.key() {
                "Esc" | "Escape" => Some(Msg::Cancel),
                _ => None,
            });
```

Unfortunately, I can't remember why I used `onkeyup` for all the cancel-on-escape features. Re-reading [this StackOverflow][3], I think it's to ensure proper ordering, so clearing the pending value happens after processing the corresponding input event. But maybe using `prevent_default` prevents the input event? I'm not sure (I'm normally not a UI dev), if you do know, just let me know. One of the replies on the same StackOverflow says that 'escape' is usually processed in `onkeydown` because of weirdness in how some browsers handle key up events for inputs and text areas, so if this is working correctly with `onkeydown,` I guess you should just keep using `onkeydown`.

[3]: https://stackoverflow.com/questions/3396754/onkeypress-vs-onkeyup-and-onkeydown

Reply:
That's fine! I appreciate your help and the opportunity to learn. 😄 

It hadn't even occurred to me to see how this was being done elsewhere in the codebase. 🤦‍♂️ 

I think it's better to be consistent with the rest of the codebase. `onkeyup` seems to be working fine. It can easily be changed if a reason does arise.